src/fsharp/FSharp.Build/Fsc.fs

```diff
@@ -198,17 +199,23 @@ type public Fsc () as this =
         // WarningLevel
         builder.AppendSwitchIfNotNull("--warn:", warningLevel)
 
+        // warnOn
+        let warnOnArray =
+            match warnOn with
+            | null -> [||]
+            | _ -> warnOn.Split([|' '; ';'; ','|], StringSplitOptions.RemoveEmptyEntries)
+
+        builder.AppendSwitchIfNotNull("--warnon:", warnOnArray, ",")
+
         // TreatWarningsAsErrors
         if treatWarningsAsErrors then
             builder.AppendSwitch("--warnaserror")
 
         // WarningsAsErrors
-        // Change warning 76, HashReferenceNotAllowedInNonScript/HashDirectiveNotAllowedInNonScript/HashIncludeNotAllowedInNonScript, into an error
-        // REVIEW: why is this logic here? In any case these are errors already by default!
         let warningsAsErrorsArray =
             match warningsAsErrors with
-            | null -> [|"76"|]
-            | _ -> (warningsAsErrors + " 76 ").Split([|' '; ';'; ','|], StringSplitOptions.RemoveEmptyEntries)
+            | null -> [||]
+            | _ -> warningsAsErrors.Split([|' '; ';'; ','|], StringSplitOptions.RemoveEmptyEntries)
```

Comment:
Refactor this to call through the same path as above and add tests so we don't have to debug this mentally.

Reply:
Done